nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -686,6 +687,42 @@ impl BundleCollection {
         )
         .await?;
 
+        // Collect reconfigurator state
+        const NMAX_BLUEPRINTS: usize = 25;
+        match reconfigurator_state_load(
+            &self.opctx,
+            &self.datastore,
+            NMAX_BLUEPRINTS,
+        )
+        .await
+        {
+            Ok(state) => {
+                let json = serde_json::to_string_pretty(&state).with_context(
+                    || "failed to serialize reconfigurator state",
+                )?;
+                tokio::fs::write(
+                    dir.path().join("reconfigurator_state.json"),
+                    json,
+                )
+                .await
+                .with_context(|| "failed to write reconfigurator_state.json")?;
```

Comment:
I wonder if it would make this meaningfully faster to use `serde_json::to_writer_pretty` with a buffered writer.  In the past I've been surprised how big a difference the buffering makes on reading (like an order of magnitude).  I'm not sure this matters but if it would let us collect more data that seems useful.  What do you think?

Reply:
`tokio::fs::write` is definitely not buffered. I think a rule of thumb is that for anything below 2MB or so, just serializing to memory and then writing that out in one go is going to be faster. (The reconfigurator state probably exceeds that, so a buffered write makes sense.)